### PR TITLE
fix(afdko): update dependencies

### PR DESCRIPTION
**Pull request by Zenvie**

### CODE REVIEW

File: afdko/.SRCINFO
Comment (alerque) [inline afdko/.SRCINFO]: Please don't keep bumping the pkgrel. Obviously you many need to bump it locally once to test, and I think we can also bump it *once* in this PR, but I'm not going to skip from 1 to 6 and the only releases that count are the ones in the AUR ... until this merges that's currently at 1.

File: afdko/.SRCINFO
Comment (alerque) [inline afdko/.SRCINFO]: Nope, this is going to be wrong. We should never need virtualenv to build a package.

File: afdko/PKGBUILD
Comment (alerque) [inline afdko/PKGBUILD]: Nice fix. Lets just use the quote style in the rest of the PKGBUILD. It's a fixed string with no variable replacements and no spaces:

```suggestion
license=(Apache-2.0)
```

File: afdko/PKGBUILD
Comment (alerque) [inline afdko/PKGBUILD]: This is unnecessary clutter. You only need to set the output filename if it is *different* from what is being downloaded. In this case it is not.

File: afdko/PKGBUILD
Comment (alerque) [inline afdko/PKGBUILD]: It is completely pointless to test packages in a virtualenv, this only tests the upstream test suite in isolation. The only point of a `check()` step in the build is to test *integration* with the host environment. If we can't work that out then there is no point in even running tests. Although of course it would be good to work out and run as many integrated tests as we can.

File: python-ufonormalizer/.SRCINFO
Comment (alerque) [inline python-ufonormalizer/.SRCINFO]: This is incorrect, the upstream canonical URL still has a capital.

File: python-ufonormalizer/PKGBUILD
Comment (alerque) [inline python-ufonormalizer/PKGBUILD]: Please don't change the coding style just for the sake of changing it. This is a fixed string with no substitutions, so single quotes are good.

File: python-ufonormalizer/PKGBUILD
Comment (alerque) [inline python-ufonormalizer/PKGBUILD]: Again, it is not necessary to rename the output file whet it already has the right name.

File: python-ufonormalizer/PKGBUILD
Comment (alerque) [inline python-ufonormalizer/PKGBUILD]: Please don't change the coding style just to change things. Tabs for indentation, and the short flags are my preference. And I diff packages with hundreds of others I maintain and expect to be able to grep them for things the way I usually code them, so changing it up hinders my workflow. Sorry.

(The change to calling pytest directly in this block is fine although I'm not sure it serves any purpose really.)

**Review by alerque — CHANGES_REQUESTED** [review]
Thanks for taking the time to contribute. The switch to PEP517 build tooling is welcome, as are any real dependency fixes. But it's a bit hard to evaluate with all the unnecessary and ever outright wrong changes thrown in. Let's strip this down to just the actual fixes and leave out style changes and get rid of pip/virtualenv stuff. Then I'll check the actual build...

File: afdko/PKGBUILD
Comment (Zenvie) [reply afdko/PKGBUILD]: https://github.com/adobe-type-tools/afdko/issues/1739#issuecomment-2016672006

Comment (alerque) [reply afdko/PKGBUILD]: The `check()` stage runs after the build but without the final package being installed, so you do have to work out all the paths. System dependencies should be installed (via `makedepends`+`checkdepends`), so just the path to internal stuff needs fixing. Usually that's what adding the build directory to `PYTHON_PATH` accomplishes, but as my commented code there suggests upstream is doing vendoring and it is not trivial to untangle what they've done with tests.

Comment (Zenvie) [reply afdko/PKGBUILD]: Must we bypass this check function? I’m aware that my current method of verification is quite inelegant.

Comment (alerque) [reply afdko/PKGBUILD]: It's not just inelegant, it doesn't actually test integration with the package. It just sets up a shim between the host system, the build, and the test suite that automatically makes up any differences. Anything missing or incompatible from the host system is just going to be supplied by the virtualenv. That means it doesn't actually reveal any issues of the sort we're looking for by having a check step in the first place.

Either we work it out so we can actually run the tests on the host system such that they serve as integration tests or we do nothing it all. There is no point in faking it and just running the upstream project's unit tests on our end. We presume they are doing that upstream anyway.

File: afdko/PKGBUILD
Comment (alerque) [inline afdko/PKGBUILD:26]: How did you confirm this is an actual direct dependency? As a policy Arch packages don't rely on transitive dependencies and do explicitly list dependencies, but that only applies if they are *used* directly. In this case I thought this was only being *used* transitively through python-lxml and hence the dependency only needed to be transitive as well. Did that change?

File: afdko/PKGBUILD
Comment (alerque) [inline afdko/PKGBUILD]: My preference is one per line except the PEP517 stuff which isn't specific to the project.

```suggestion
             python-scikit-build
             python-setuptools-scm)
```

File: afdko/PKGBUILD
Comment (skef) [reply afdko/PKGBUILD:26]: The c/c++ code embedded in the AFDKO package does link against libxml2. That's been true for quite a while -- I believe more than a year now.

Comment (Zenvie) [reply afdko/PKGBUILD:26]: The error was identified by namcap.

File: python-ufonormalizer/PKGBUILD
Comment (alerque) [inline python-ufonormalizer/PKGBUILD]: This bit doesn't seem right. It's possible for setuptools to be a runtime dependency (is it actually in this case) but it is almost impossible for the scm variant to be needed at runtime. If this actually does need setuptools at runtime, lets set setuptools as the dependency and keep setuptools-scm in the build time only dependency list.

**Review by alerque — CHANGES_REQUESTED** [review]

File: afdko/PKGBUILD
Comment (alerque) [reply afdko/PKGBUILD:26]: Thanks for the info. As the current `namcap` maintainer I can say with confidence it does not always get this right, but yes if the AFDKO code does link libxml2 this is correct, thanks.

File: python-ufonormalizer/PKGBUILD
Comment (Zenvie) [reply python-ufonormalizer/PKGBUILD]:
```log
Checking python-ufonormalizer-0.6.1-4-any.pkg.tar.zst
python-ufonormalizer E: Dependency python-setuptools-scm detected and not included (python modules ['setuptools_scm.get_version'] needed in files ['usr/lib/python3.11/site-packages/ufonormalizer/__init__.py'])
python-ufonormalizer W: Dependency included, but may not be needed ('python-setuptools')
```

Comment (alerque) [reply python-ufonormalizer/PKGBUILD]: Did you look at the script itself? Namcap does some simple grep-like searches but doesn't actually follow along with Python logic. This line is in a `try:` statement that is allowed to fail. And actually in this case it should never even *reach* this line of code because the version should be generated by the build. The code is only really useful for running an un-built/un-installed copy of the source.

**Review by alerque — CHANGES_REQUESTED** [review]

File: python-ufonormalizer/PKGBUILD
Comment (alerque) [inline python-ufonormalizer/PKGBUILD]: I just looked and I don't see any reason to make this a runtime dependency. Where are you getting this from?

**Review by alerque — CHANGES_REQUESTED** [review]
Almost there. This is looking pretty good but I still don't see why setuptools is being made a run time dependency.

**Pull request merged**
